pkg/detectors/meraki/meraki.go

```diff
@@ -0,0 +1,115 @@
+	"strings"
+
+	regexp "github.com/wasilibs/go-re2"
+
+	"github.com/trufflesecurity/trufflehog/v3/pkg/common"
+	"github.com/trufflesecurity/trufflehog/v3/pkg/detectors"
+	"github.com/trufflesecurity/trufflehog/v3/pkg/pb/detectorspb"
+)
+
+type Scanner struct {
+	client *http.Client
+}
+
+// Ensure the Scanner satisfies the interface at compile time.
+var _ detectors.Detector = (*Scanner)(nil)
+
+var (
+	defaultClient = common.SaneHttpClient()
+
+	apiKey = regexp.MustCompile(detectors.PrefixRegex([]string{"meraki"}) + `([0-9a-f]{40})`)
+)
+
+func (s Scanner) getClient() *http.Client {
+	if s.client != nil {
+		return s.client
+	}
+
+	return defaultClient
+}
+
+// Keywords are used for efficiently pre-filtering chunks.
+// Use identifiers in the secret preferably, or the provider name.
+func (s Scanner) Keywords() []string {
+	return []string{"meraki"}
+}
+
+func (s Scanner) Description() string {
+	return "Cisco Meraki is a cloud-managed IT solution that provides networking, security, and device management through an easy-to-use interface." +
+		"Meraki APIs make it possible to rapidly deploy and manage networks at scale, build on a platform of intelligent, cloud-connected IT products, and engage with users in powerful new ways."
+}
+
+// FromData will find and optionally verify Meraki API Key secrets in a given set of bytes.
+func (s Scanner) FromData(ctx context.Context, verify bool, data []byte) (results []detectors.Result, err error) {
+	dataStr := string(data)
+
+	// uniqueMatches will hold unique match values and ensure we only process unique matches found in the data string
+	var uniqueMatches = make(map[string]bool)
+
+	for _, match := range apiKey.FindAllStringSubmatch(dataStr, -1) {
+		if len(match) != 2 {
+			continue
+		}
+
+		uniqueMatches[strings.TrimSpace(match[1])] = true
+	}
+
+	for match := range uniqueMatches {
+		s1 := detectors.Result{
+			DetectorType: detectorspb.DetectorType_Meraki,
+			Raw:          []byte(match),
+		}
+
+		if verify {
+			client := s.getClient()
+			isVerified, verificationErr := verifyMerakiApiKey(ctx, client, match)
+			s1.Verified = isVerified
+			if verificationErr != nil {
+				s1.SetVerificationError(verificationErr)
+			}
+		}
+
+		results = append(results, s1)
+	}
+
+	return results, nil
+}
+
+func (s Scanner) Type() detectorspb.DetectorType {
+	return detectorspb.DetectorType_Meraki
+}
+
+/*
+verifyMerakiApiKey verifies if the passed matched api key for meraki is active or not.
+docs: https://developer.cisco.com/meraki/api-v1/authorization/#authorization
+*/
+func verifyMerakiApiKey(ctx context.Context, client *http.Client, match string) (bool, error) {
+	req, err := http.NewRequestWithContext(ctx, "GET", "https://api.meraki.com/api/v1/organizations", http.NoBody)
+	if err != nil {
+		return false, err
+	}
+
+	// set the required auth header
+	req.Header.Set("X-Cisco-Meraki-API-Key", match)
+
+	resp, err := client.Do(req)
+	if err != nil {
+		return false, err
+	}
+	defer resp.Body.Close()
```

Comment:
It's ostensibly 'best' to manually drain the response body. https://github.com/trufflesecurity/trufflehog/pull/2086#issue-1974701062

```suggestion
        defer func() {
	  _, _ = io.Copy(io.Discard, res.Body)
	  _ = res.Body.Close()
        }()
```